connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedConfig.java

```diff
@@ -38,6 +39,13 @@
     public static final String GROUP_ID_CONFIG = "group.id";
     private static final String GROUP_ID_DOC = "A unique string that identifies the Connect cluster group this worker belongs to.";
 
+    /**
+     * <code>group.instance.id</code>
+     */
+    public static final String GROUP_INSTANCE_ID_CONFIG = "group.instance.id";
```

Comment:
Hmm.. I think I missed this addition in the KIP. How much effort would it be to pull this change into a separate PR? I think we may need some discussion.

Reply:
Why couldn't we piggy-back the change in this PR? Connect could also benefit from using static membership right.

Reply:
I agree with @hachikuji this is too significant to be omitted from a KIP and we should probably avoid piggybacking such a change in a subtle way in this already big PR.

The doc of the config below is indicative that we need to give this more thought. The connect worker is not a consumer and it doesn't use the group membership protocol in the same way. This is even more true with the changes being introduced soon with incremental cooperative rebalancing in Connect. The interplay between KIP-415 and static membership has not been sufficiently studied yet and therefore I'd suggest not introducing everything at once with the upcoming release.

Reply:
@kkonstantine @hachikuji sounds great! My original thought was that the change happens on abstract coordinator layer, so consequently we could cover all the subclass use cases (both consumer and connect). I will revert connect related changes.

Reply:
Possibly so, though I am not sure since it does not local state like Streams. In any case, I do not want to see this PR blocked by this discussion, so my thought was to split it into a separate PR.

cc @rhauch Any thoughts about this?

Reply:
Apologies for the late comment above. I hadn't refreshed the page and seen the updates.

Reply:
It's fine :)